src/Unleash/Communication/UnleashApiClient.cs

```diff
@@ -190,7 +190,6 @@ public async Task<bool> RegisterClient(ClientRegistration registration, Cancella
             using (var request = new HttpRequestMessage(HttpMethod.Post, requestUri))
             {
                 request.Content = new StringContent(JsonSerializer.Serialize(registration, options), Encoding.UTF8, "application/json");
```

Comment:
Needs to be set in the request content itself, otherwise it doesn't materialise as json on the receiver